transport/transporters/kafka/batch/batch.go

```diff
@@ -0,0 +1,114 @@
+package batch
```

Comment:
at first I was considering adding a variadic option to `GenericBatch` to optionally accept a maxByteSize per message, but I instead created a custom batch for Kafka.

Reply:
`KafkaBatch` basically does all the same things as `GenericBatch` but also
- checks for message byte size and adds a 
- creates the ProduceMessage from the `MarshalledMessage`